lib/datasource/terraform-provider/index.ts

```diff
@@ -116,4 +123,114 @@ export class TerraformProviderDatasource extends TerraformDatasource {
     logger.trace({ dep }, 'dep');
     return dep;
   }
+
+  @cache({
+    namespace: `datasource-${TerraformProviderDatasource.id}-builds`,
+    key: (registryURL: string, repository: string, version: string) =>
+      `${registryURL}/${repository}/${version}`,
+  })
+  async getBuilds(
+    registryURL: string,
+    repository: string,
+    version: string
+  ): Promise<TerraformBuild[]> {
+    if (registryURL === TerraformProviderDatasource.defaultRegistryUrls[1]) {
+      // check if registryURL === secondary backend
+      const repositoryRegexResult =
+        TerraformProviderDatasource.repositoryRegex.exec(repository);
+      if (!repositoryRegexResult) {
+        // non hashicorp builds are not supported with releases.hashicorp.com
+        return null;
+      }
+      const lookupName = repositoryRegexResult.groups.lookupName;
+      const backendLookUpName = `terraform-provider-${lookupName}`;
+      let versionReleaseBackend: VersionDetailResponse;
+      try {
+        versionReleaseBackend = await this.getReleaseBackendIndex(
+          backendLookUpName,
+          version
+        );
+      } catch (err) {
+        logger.debug(
+          { err, backendLookUpName, version },
+          `Failed to retrieve builds for ${backendLookUpName} ${version}`
+        );
+        throw err;
```

Comment:
```suggestion
        if (err instanceof ExternalHostError){
          throw err;
        }
        logger.debug(
          { err, backendLookUpName, version },
          `Failed to retrieve builds for ${backendLookUpName} ${version}`
        );
        return null;
```

Only rethrow `ExternalHostError` 😉  Sorry when i wasn't clear enough